tests/http_server_test.cc

```diff
@@ -855,3 +864,221 @@ TEST(http_server_test, client_multiple_requests_disconnects_handled)
+            writer.send(Http::Code::Bad_Request);
+            return;
+        }
+
+        // Get the client body...
+        const auto client_body = request.body();
+
+        // Compress differently, depending on requested encoding...
+        switch(request.headers().get<Http::Header::AcceptEncoding>().get()->encoding())
+        {
+#ifdef PISTACHE_USE_CONTENT_ENCODING_DEFLATE
+            case Http::Header::Encoding::Deflate:
+            {
+                // Enable deflate compression...
+                writer.setCompression(Http::Header::Encoding::Deflate);
+
+                // Set maximum compression...
+                writer.setCompressionDeflateLevel(Z_BEST_COMPRESSION);
+
+                // Notify client to expect deflate compressed response...
+                writer.headers().add<Http::Header::ContentEncoding>(
+                    Http::Header::Encoding::Deflate);
+
+                // Send compressed response of original client body...
+                writer.send(Http::Code::Ok, client_body);
+
+                // Done...
+                break;
+            }
+#endif
+
+            // Unsupported encoding requested...
+            default:
+            {
+                writer.send(Http::Code::Bad_Request);
+                break;
+            }
+        }
+    }
+};
+
+#ifdef PISTACHE_USE_CONTENT_ENCODING_DEFLATE
+TEST(http_server_test, server_with_content_encoding_deflate)
+{
+    // Data to send to server to expect it to return compressed...
+
+        // Allocate storage...
+        std::vector<std::byte> originalUncompressedData(1024);
+
+        // Random bytes engine...
+        using random_bytes_engine_type =
+            std::independent_bits_engine<
+                std::default_random_engine, CHAR_BIT, unsigned char>;
+        random_bytes_engine_type randomEngine;
+
+        // Fill with random bytes...
+        std::generate(
+            std::begin(originalUncompressedData),
+            std::end(originalUncompressedData),
+            [&randomEngine]() { return static_cast<std::byte>(randomEngine()); });
+
+    // Bind server to localhost on a random port...
+    const Pistache::Address address("localhost", Pistache::Port(0));
+
+    // Initialize server...
+    Http::Endpoint server(address);
+    auto flags       = Tcp::Options::ReuseAddr;
+    auto server_opts = Http::Endpoint::options().flags(flags);
+    server_opts.maxRequestSize(1024 * 1024 * 20);
+    server_opts.maxResponseSize(1024 * 1024 * 20);
+    server.init(server_opts);
+    server.setHandler(Http::make_handler<ContentEncodingHandler>());
+    server.serveThreaded();
+
+    // Verify server is running...
+    ASSERT_TRUE(server.isBound());
+
+    // Log server coordinates...
+    const std::string server_address = "localhost:" + server.getPort().toString();
+    LOGGER("test", "Server address: " << server_address);
+
+    // Initialize client...
+
+        // Construct and initialize...
+        Http::Experimental::Client client;
```

Comment:
Just wanted to say thanks for using our own http client for writing tests, so that we're more incentivized in making it better :)

Reply:
It was a pain in the ass to figure out and it does have some weird quirks, but we got it working at least in the unit test.